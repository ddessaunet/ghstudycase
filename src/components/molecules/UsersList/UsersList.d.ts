import { User as IUser } from 'services/UserService.d';

export interface Props {
  users: IUser[];
}
