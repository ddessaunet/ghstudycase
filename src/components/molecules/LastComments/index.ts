export type { Props as LastCommentsProps } from './LastComments.d';
export { LastComments } from './LastComments';
