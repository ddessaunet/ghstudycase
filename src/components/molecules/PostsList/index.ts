export type { Props as PostsListProps } from './PostsList.d';
export { PostsList } from './PostsList';
