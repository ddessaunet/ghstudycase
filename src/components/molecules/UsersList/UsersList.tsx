import React from 'react';
import { useNavigate } from 'react-router-dom';
import { Button, Flex } from '@chakra-ui/react';
import { User } from 'components/atoms/User';
import { getPostsByUserPayload } from 'services/UserService';

export const UsersList = ({ users }: any) => {
  const navigate = useNavigate();
  const nav = (path: string) => {
    window.scrollTo(0, 0);
    navigate(path);
  };
  return (
    <Flex flexDirection="column">
      {users.map((user: any) => (
        <Flex key={user.id} m="10px">
          <User {...user} request={() => getPostsByUserPayload(user.id, 1, 0)}>
            <Flex flexDirection="column">
              <Flex gap="1">
                <Button onClick={() => nav(`/${user.id}`)}>
                  Show full profile
                </Button>
                <Button onClick={() => nav(`/${user.id}/posts`)}>
                  User posts
                </Button>
              </Flex>
            </Flex>
          </User>
        </Flex>
      ))}
    </Flex>
  );
};
