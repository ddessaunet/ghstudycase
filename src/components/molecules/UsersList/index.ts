export type { Props as UsersListProps } from './UsersList.d';
export { UsersList } from './UsersList';
