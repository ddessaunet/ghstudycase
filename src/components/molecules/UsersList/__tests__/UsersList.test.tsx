import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { UsersList, UsersListProps } from 'components/molecules/UsersList';

describe('<UsersList />', () => {
  const props = {
    users: [
      {
        id: '60d0fe4f5311236168a109ca',
        title: 'ms',
        firstName: 'Sara',
        lastName: 'Andersen',
        picture: 'https://randomuser.me/api/portraits/women/58.jpg',
      },
      {
        id: '60d0fe4f5311236168a109cb',
        title: 'miss',
        firstName: 'Edita',
        lastName: 'Vestering',
        picture: 'https://randomuser.me/api/portraits/med/women/89.jpg',
      },
    ],
  };
  const renderComponent = (props: UsersListProps) =>
    render(
      <MemoryRouter>
        <ChakraProvider>
          <UsersList {...props} />
        </ChakraProvider>
      </MemoryRouter>,
    );

  it('should render and match snapshot', () => {
    const component = renderComponent(props);
    expect(component).toMatchSnapshot();
  });
});
