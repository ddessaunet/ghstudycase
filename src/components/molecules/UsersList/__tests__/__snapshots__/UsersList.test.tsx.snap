// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersList /> should render and match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-j7qwjs"
      >
        <div
          class="css-cocn0m"
        >
          <div
            class="chakra-container css-1epz950"
          >
            <div
              class="css-obc5ee"
            >
              <div
                class="css-1juxs9n"
              >
                <img
                  alt=""
                  class="chakra-image css-1nbic85"
                  src="https://randomuser.me/api/portraits/women/58.jpg"
                />
              </div>
              <div
                class="css-18l19nf"
              >
                <div
                  class="css-1x1m52h"
                >
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-ty7r4z"
                    >
                      ms
                    </p>
                  </div>
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-ty7r4z"
                    >
                      Sara
                    </p>
                  </div>
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-ty7r4z"
                    >
                      Andersen
                    </p>
                  </div>
                </div>
                <p
                  class="chakra-text css-1tx38fa"
                >
                  Total posts: 
                  Loading...
                </p>
                <div
                  class="css-j7qwjs"
                >
                  <div
                    class="css-1frb2eh"
                  >
                    <button
                      class="chakra-button css-taj3dd"
                      type="button"
                    >
                      Show full profile
                    </button>
                    <button
                      class="chakra-button css-taj3dd"
                      type="button"
                    >
                      User posts
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-cocn0m"
        >
          <div
            class="chakra-container css-1epz950"
          >
            <div
              class="css-obc5ee"
            >
              <div
                class="css-1juxs9n"
              >
                <img
                  alt=""
                  class="chakra-image css-1nbic85"
                  src="https://randomuser.me/api/portraits/med/women/89.jpg"
                />
              </div>
              <div
                class="css-18l19nf"
              >
                <div
                  class="css-1x1m52h"
                >
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-ty7r4z"
                    >
                      miss
                    </p>
                  </div>
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-ty7r4z"
                    >
                      Edita
                    </p>
                  </div>
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-ty7r4z"
                    >
                      Vestering
                    </p>
                  </div>
                </div>
                <p
                  class="chakra-text css-1tx38fa"
                >
                  Total posts: 
                  Loading...
                </p>
                <div
                  class="css-j7qwjs"
                >
                  <div
                    class="css-1frb2eh"
                  >
                    <button
                      class="chakra-button css-taj3dd"
                      type="button"
                    >
                      Show full profile
                    </button>
                    <button
                      class="chakra-button css-taj3dd"
                      type="button"
                    >
                      User posts
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-j7qwjs"
    >
      <div
        class="css-cocn0m"
      >
        <div
          class="chakra-container css-1epz950"
        >
          <div
            class="css-obc5ee"
          >
            <div
              class="css-1juxs9n"
            >
              <img
                alt=""
                class="chakra-image css-1nbic85"
                src="https://randomuser.me/api/portraits/women/58.jpg"
              />
            </div>
            <div
              class="css-18l19nf"
            >
              <div
                class="css-1x1m52h"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-ty7r4z"
                  >
                    ms
                  </p>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-ty7r4z"
                  >
                    Sara
                  </p>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-ty7r4z"
                  >
                    Andersen
                  </p>
                </div>
              </div>
              <p
                class="chakra-text css-1tx38fa"
              >
                Total posts: 
                Loading...
              </p>
              <div
                class="css-j7qwjs"
              >
                <div
                  class="css-1frb2eh"
                >
                  <button
                    class="chakra-button css-taj3dd"
                    type="button"
                  >
                    Show full profile
                  </button>
                  <button
                    class="chakra-button css-taj3dd"
                    type="button"
                  >
                    User posts
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-cocn0m"
      >
        <div
          class="chakra-container css-1epz950"
        >
          <div
            class="css-obc5ee"
          >
            <div
              class="css-1juxs9n"
            >
              <img
                alt=""
                class="chakra-image css-1nbic85"
                src="https://randomuser.me/api/portraits/med/women/89.jpg"
              />
            </div>
            <div
              class="css-18l19nf"
            >
              <div
                class="css-1x1m52h"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-ty7r4z"
                  >
                    miss
                  </p>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-ty7r4z"
                  >
                    Edita
                  </p>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-ty7r4z"
                  >
                    Vestering
                  </p>
                </div>
              </div>
              <p
                class="chakra-text css-1tx38fa"
              >
                Total posts: 
                Loading...
              </p>
              <div
                class="css-j7qwjs"
              >
                <div
                  class="css-1frb2eh"
                >
                  <button
                    class="chakra-button css-taj3dd"
                    type="button"
                  >
                    Show full profile
                  </button>
                  <button
                    class="chakra-button css-taj3dd"
                    type="button"
                  >
                    User posts
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
